src/main/java/hudson/plugins/ec2/EC2HostAddressProvider.java

```diff
@@ -0,0 +1,56 @@
+package hudson.plugins.ec2;
+
+import com.amazonaws.services.ec2.model.Instance;
+import com.google.common.net.HostAndPort;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Optional;
+
+import static hudson.plugins.ec2.ConnectionStrategy.*;
+
+public class EC2HostAddressProvider {
```

Comment:
You can make this class `final` with a `private` constructor since it's all static methods. Unless you intend to make this class overridable in general?